optuna/visualization/matplotlib/_intermediate_values.py

```diff
@@ -1,11 +1,72 @@
 from optuna._experimental import experimental
+from optuna.logging import get_logger
+from optuna.study import Study
+from optuna.trial import TrialState
 from optuna.visualization.matplotlib._matplotlib_imports import _imports
 
 
 if _imports.is_successful():
     from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
+
+_logger = get_logger(__name__)
 
 
 @experimental("2.2.0")
-def plot_intermediate_values() -> Axes:
-    raise NotImplementedError("To be implemented soon.")
+def plot_intermediate_values(study: Study) -> "Axes":
+    """Plot intermediate values of all trials in a study with Matplotlib.
+
+    .. seealso::  optuna.visualization.plot_intermediate_values
```

Comment:
This `.. seealso::` section can be made more descriptive.
```suggestion
    .. seealso::  
        Please refer to :func:`~optuna.visualization.plot_intermediate_values` for an example.
```